src/connectors/panoptic/panoptic.ts

```diff
@@ -286,632 +302,575 @@ export class Panoptic {
     try {
       const panopticHelperAddress = this.PanopticHelper;
       const panopticHelperContract = new Contract(panopticHelperAddress, panopticHelperAbi.abi, wallet);
-      const tokenId = await panopticHelperContract.createZEEHBS(
-        univ3pool, 
-        width, 
-        longStrike, 
-        shortStrike, 
+      return await panopticHelperContract.createZEEHBS(
+        univ3pool,
+        width,
+        longStrike,
+        shortStrike,
         asset,
         ratio
-      )
-      return tokenId
-    } catch (error) {
-      return error;
-    }
-  }
-
-  // TODO: These methods on the helper aren't live yet; eventually, this will be the easier way
-  //       to get multiple greeks in one call.
-  async queryGreeks(
-    wallet: Wallet,
-    tick: number,
-    positionIdList: BigNumber[],
-    greek: string
-  ): Promise<any> {
-    try {
-      const panopticHelper = this.PanopticHelper;
-      const panopticHelperContract = new Contract(panopticHelper, panopticHelperAbi.abi, wallet);
-      let greekValue;
-      if (greek === "delta") {
-        greekValue = await panopticHelperContract.delta(this.PanopticPool, wallet.address, tick, positionIdList);
-      } else if (greek === "gamma") {
-        greekValue = await panopticHelperContract.gamma(this.PanopticPool, wallet.address, tick, positionIdList);
-      } else {
-        throw new Error("Invalid greek");
-      }
-      logger.info(`PanopticHelper reports ${greek} = ${greekValue}`);
-      return greekValue;
+      );
     } catch (error) {
-      logger.error(`Error checking ${greek}...`, error);
-      return error;
+      return new Error("Error calculating createZEEHBS: " + (error as Error).message);
     }
   }
 
   // PanopticPool interactions
-
   async calculateAccumulatedFeesBatch(
     wallet: Wallet,
     includePendingPremium: boolean = false,
     positionIdList: BigNumber[]
-  ): Promise<any> {
+  ): Promise<{ "premium0": BigNumber, "premium1": BigNumber, [key: number]: BigNumber } | Error> {
     try {
       const panopticpool = this.PanopticPool;
-      logger.info(`Checking calculateAccumulatedFeesBatch...`)
       const panopticPoolContract = new Contract(panopticpool, panopticPoolAbi.abi, wallet);
-      const data = await panopticPoolContract.calculateAccumulatedFeesBatch(
+      const result = await panopticPoolContract.calculateAccumulatedFeesBatch(
         wallet.address,
         includePendingPremium,
         positionIdList
       );
-      return data;
+      const { premium0, premium1, ...rest } = result;
+      return { premium0, premium1, ...rest };
     } catch (error) {
-      logger.error("Error on calculateAccumulatedFeesBatch:", error);
-      return error;
+      return new Error("Error on calculateAccumulatedFeesBatch: " + (error as Error).message);
     }
   }
 
   async collateralToken0(
     wallet: Wallet
-  ): Promise<any> {
+  ): Promise<{"collateralToken": BigNumber} | Error> {
```

Comment:
I'm surprised addresses come back as BigNumbers - but i assume you were able to consume this endpoint in python and still wrap the returned address into a Contract object and use it?

Reply:
Good catch, we should check this to make sure it still works. I see that an address is returned - if it comes back as a uint, it should be fine. I'm going to move this to a separate issue, merge the PR, and circle back to it.